File: frontend/src/services/appointmentService.ts
import { api } from '@/lib/api'
import type { Appointment, PaginatedResponse } from '@/types'

export interface AppointmentPayload {
  incident_id?: number
  farm_id?: number
  technician_id?: number
  farmer_id?: number
  scheduled_at: string
  purpose?: string
  notes?: string
}

export interface AppointmentTechnician {
  id: number
  full_name: string
  phone?: string
  assigned?: boolean
}

export const appointmentService = {
  list: (status?: string) => api.get<PaginatedResponse<Appointment>>('/appointments', { params: { status } }),
  listTechnicians: () => api.get<{ data: AppointmentTechnician[] }>('/farmer/technicians'),
  create: (payload: AppointmentPayload) => api.post<{ message: string; data: Appointment }>('/appointments', payload),
  updateStatus: (id: number, status: Appointment['status']) =>
    api.put<{ message: string; data: Appointment }>(`/appointments/${id}/status`, { status }),
}

File: mobile/lib/services/appointment_service.dart
import '../core/api/api_client.dart';
import '../models/appointment.dart';

class AppointmentService {
  AppointmentService(this._api);

  final ApiClient _api;

  Future<List<Appointment>> list({String? status}) {
    return _api.handle(
      _api.get('/appointments', queryParameters: {
        if (status != null && status.isNotEmpty) 'status': status,
      }),
      (json) {
        final data = (json as Map<String, dynamic>)['data'] as List<dynamic>;
        return data.map((e) => Appointment.fromJson(e as Map<String, dynamic>)).toList();
      },
    );
  }

  Future<Appointment> create({
    required int technicianId,
    required String scheduledAt,
    String? purpose,
    String? notes,
    int? incidentId,
    int? farmId,
  }) {
    return _api.handle(
      _api.post('/appointments', data: {
        'technician_id': technicianId,
        'scheduled_at': scheduledAt,
        if (purpose != null && purpose.isNotEmpty) 'purpose': purpose,
        if (notes != null && notes.isNotEmpty) 'notes': notes,
        if (incidentId != null) 'incident_id': incidentId,
        if (farmId != null) 'farm_id': farmId,
      }),
      (json) => Appointment.fromJson((json as Map<String, dynamic>)['data'] as Map<String, dynamic>),
    );
  }

  Future<void> updateStatus(int id, String status) {
    return _api.handle(
      _api.put('/appointments/$id/status', data: {'status': status}),
      (_) {},
    );
  }
}
